### DISCUSSION TITLE
Configure renovate to update specific packages and group them by path in monorepo

### DISCUSSION THREAD

**Discussion by alaamub** [renovatebot/renovate#12748] [Request Help]
We have a monorepo where we want to configure renovatebot to only issue updates to specific list of `allowlisted packages` per subpath within the monorepo per specific files, for example, only update `numpy` and `urllib3` across all these subpaths `["project1/", "project2/"]` and only update specific files within these subprojects then group the updates according to the project name. 

for example the first PR will be updating `numpy` and `urllib3` for `project1` in `requirements.in` and group them, then another PR for updating `numpy` and `urllib3` for `project2` for `requirements.in` and `requirements.txt` and group them etc. 

Also, ignore everything else, meaning if you see `numpy` in `project1` in file called `default.txt` then don't do that. 

It is working as expected, but I have to specify all files names all the time in both `includePaths` and `matchPaths` otherwise, renovate will create additional PRs for other files. What is the best configuration to achieve the above requirements? thanks

<details><summary>Click me to see config</summary>

```
{
  "repositories": ["org/monorepo"],
  "includePaths": [
    "project1/data/requirements/requirements.in",
    "project2/venv/requirements.in",
    "project2/venv/requirements.txt"
  ],
  "updateLockFiles": false,
  "lockFileMaintenance": {
    "enabled": false
  },
  "enabledManagers": ["pip_requirements"],
  "pip_requirements": {
    "enabled": true
  },
  "packageRules": [
    {
      "matchUpdateTypes": ["major"],
      "enabled": false
    },
    {
      "matchLanguages": ["python"]
    },
    {
      "packagePatterns": ["*"],
      "excludePackageNames": [
        "numpy",
        "urllib3"
      ],
      "enabled": false
    },
    {
      "paths": ["project1/**"],
      "matchPaths": ["project1/data/requirements/requirements.in"],
      "matchPackageNames": [
        "numpy",
        "urllib3"
      ],
      "matchUpdateTypes": ["minor", "patch"],
      "groupName": "dependencies for project1"
    },
    {
      "paths": ["project2/**"],
      "matchPackageNames": [
        "numpy",
        "urllib3"
      ],
      "matchPaths": ["venv/requirements.txt", "venv/requirements.in"],
      "matchUpdateTypes": ["minor", "patch"],
      "groupName": "dependencies for project2"
    }
  ],
  "platform": "github",
  "vulnerabilityAlerts": {
    "enabled": false
  },
  "rebaseStalePrs": true,
  "pinDigests": true,
  "dryRun": false,
  "autodiscover": false,
  "onboardingConfigFileName": ".renovaterc.json"
}
```
</details>

**Comment by rarkins** [comment]
You can also use additionalBranchPrefix combined with parentDir. Please search this repo discussions for these terms as you can find some previous examples

**Reply by alaamub** [reply]
Thanks @rarkins this would create seperate PRs and append `baseDir` to the commitMessageSuffix and additionalBranchPrefix. how to make sure you can group them by the  `parentDir` `"groupName": "({{baseDir}})"` does not work?

```
{
      "managers": ["pip_requirements"],
      "additionalBranchPrefix": "{{baseDir}}-",
      "packagePatterns": [".*"],
      "commitMessageSuffix": "({{baseDir}})"
    },
```

**Reply by rarkins** [reply]
I don't understand the question.
